cpp/src/io/parquet/page_enc.cu

```diff
@@ -1337,11 +1345,196 @@ static __device__ void byte_reverse128(__int128_t v, void* dst)
                d_char_ptr);
 }
 
-__device__ void get_extremum(const statistics_val* stats_val,
-                             statistics_dtype dtype,
-                             void* scratch,
-                             const void** val,
-                             uint32_t* len)
+/**
+ * @brief Test to see if an array contains a valid UTF-8 string.
+ *
+ * @param str device_span containing array to test.
+ * @return true if the array contains a valid UTF-8 string.
+ */
+__device__ bool is_valid_utf8_string(device_span<uint8_t const> str)
```

Comment:
UTF-8 is inherently unsigned.  Even `string_view` casts to uint8_t when dealing with it.

Reply:
I'm not trying to imply sign as much as using char instead of int8. So in this case, I would expect it to be a `device_span<unsigned char>`. This is a nit at best.

Reply:
Do/Can we use `string_view` in this context?

Reply:
> I'm not trying to imply sign as much as using char instead of int8. So in this case, I would expect it to be a `device_span<unsigned char>`. This is a nit at best.

After I replied I thought that might be where you were going.  I'm fine with u_char rather than uint8

Reply:
> Do/Can we use `string_view` in this context?

This gets back to your comment about too many ways to reference 8 bits :wink:

Reply:
Yeah, I don't really care which we pick as long as we're consistent here. Mixing `unsigned char`, `uint8_t`, and `std::byte` with variable names `c`, `str`, and `byte` can lead to confusion. If we pick one method and name variables that match it I'm fine and happy. You pick your soup de jour and I'll nod and approve. :)

Reply:
Take a look now.  Getting closer I think?  I left the `truncate_binary` using `uint8_t` since that's closer to binary than char is.  Or should i just keep them all consistent as `unsigned char`?

Reply:
I like `std::byte`, but we've been there. 😄  I think `uint8_t` is a fine representation for binary data.

Reply:
You know, if they'd allow bitops with ints/literals/enums, or enums could have std::byte type, I'd be on board. Just for fun I tried rewriting the parquet stats to use std::byte.  Wound up ditching it in many places because you need to mask off bits and compare to an enum...so much casting back and forth.  It just wasn't...you know...worth the...something. 🟠

Reply:
Now I'm sad, you didn't come through with the juice line completely here.

Reply:
:rofl: It's an inside joke now